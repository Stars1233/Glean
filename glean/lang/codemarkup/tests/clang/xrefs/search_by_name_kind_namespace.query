query: search.code.SearchByNameKindAndLanguage { name = "foo", language = Cpp, kinds = { just = Namespace } }
perf: true
transform: [gensort, []]
