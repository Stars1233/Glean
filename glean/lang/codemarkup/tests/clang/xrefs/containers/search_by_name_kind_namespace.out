[
  "@generated",
  {
    "key": {
      "name": "foo",
      "language": 0,
      "kinds": 5,
      "entity": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "foo" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 26,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "S" },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "foo" } }
                                }
                              }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 11,
                            "columnBegin": 1,
                            "lineEnd": 11,
                            "columnEnd": 19
                          }
                        }
                      }
                    },
                    {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "foo" } }
                                }
                              }
                            }
                          },
                          "kind": { "union_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 13,
                            "columnBegin": 1,
                            "lineEnd": 17,
                            "columnEnd": 1
                          }
                        }
                      }
                    },
                    {
                      "enum_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "E" },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "foo" } }
                                }
                              }
                            }
                          },
                          "isScoped": false,
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 19,
                            "columnBegin": 1,
                            "lineEnd": 19,
                            "columnEnd": 24
                          }
                        }
                      }
                    },
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": {
                                "key": {
                                  "name": { "key": "this_is_a_function" }
                                }
                              },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "foo" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "int" },
                              "parameters": [
                                {
                                  "name": { "key": "a" },
                                  "type": { "key": "int" }
                                },
                                {
                                  "name": { "key": "b" },
                                  "type": { "key": "int" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 21,
                            "columnBegin": 1,
                            "lineEnd": 24,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
]